src/Descriptor/AssemblyScanner.cs

```diff
@@ -12,10 +12,13 @@ public class AssemblyScanner : IEnumerable<IDescriptor<IDescriptorContainer>>
         public AssemblyScanner(IEnumerable<Type> scannerTypes)
         {
             this.scannerTypes = scannerTypes;
+            SetActivator(t => Activator.CreateInstance(t) as IDescriptor<IDescriptorContainer>);
         }
 
         private readonly IEnumerable<Type> scannerTypes;
 
+        public Func<Type, IDescriptor<IDescriptorContainer>> DescriptorActivator { get; set; }
```

Comment:
Having the setter on this property be public in addition to exposing a public `SetActivator` method makes this class's API a little confusing. It would be more straight forward if only one of these two options were available for setting this value.